src/libultra/io/vimgr.c

```diff
@@ -1,5 +1,116 @@
 #include "global.h"
+#include "osint.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/boot/vimgr/osCreateViManager.s")
+typedef struct {
+    u16 unk00;
+    u8 unk02;
+    u32 unk04;
+    char pad[0xC];
+    u16 unk14;
+    u16 unk16;
+} viMesgStruct;
 
-#pragma GLOBAL_ASM("asm/non_matchings/boot/vimgr/viMgrMain.s")
+OSThread viThread;
+u8 viThreadStack[0x1000];
+OSMesgQueue viEventQueue;
+OSMesg viEventBuf[6];
+viMesgStruct viRetraceMsg;
+viMesgStruct viCounterMsg;
```

Comment:
The correct type for these is `OSIoMesg` apparently
(https://github.com/Rozelette/libultra-mdebug/blob/master/vimgr.o.txt#L176)

Reply:
Hm, I wonder if there's an enum for the `type`:
```c
        viRetraceMsg.hdr.type = 13;
        viRetraceMsg.hdr.pri = 0;
        viRetraceMsg.hdr.retQueue = NULL;
        viCounterMsg.hdr.type = 14;
        viCounterMsg.hdr.pri = 0;
        viCounterMsg.hdr.retQueue = NULL;
```